crates/devolutions-pedm/src/error.rs

```diff
@@ -0,0 +1,93 @@
+use std::fmt::Debug;
+
+use aide::OperationOutput;
+use axum::{
+    response::{IntoResponse, Response},
+    Json,
+};
+use hyper::StatusCode;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use tracing::error;
+use win_api_wrappers::raw::Win32::Foundation::{
+    ERROR_ACCESS_DISABLED_BY_POLICY, ERROR_CANCELLED, ERROR_INVALID_PARAMETER, E_UNEXPECTED,
+};
+
+#[derive(Deserialize, Serialize, Debug, JsonSchema)]
+#[serde(rename_all = "PascalCase")]
+pub enum Error {
```

Comment:
maybe lets use `thiserror` here?

Reply:
In this case, I believe the benefit to be too minor to be worth it. This error is useful for deriving the `Display` trait with nice user-facing errors, but we don’t have such error messages here. See [this comment](https://github.com/Devolutions/devolutions-gateway/pull/942/files/0deddb6d97397eba1462f66fc044bff5f7d2e605#r1704158166).